src/main/java/com/jcraft/jsch/OpenSSHConfig.java

```diff
@@ -74,6 +77,9 @@
  */
 public class OpenSSHConfig implements ConfigRepository {
 
+  private static final List<String> keysWithListAdoption = Stream.of("KexAlgorithms", "Ciphers", "HostbasedAcceptedAlgorithms", 
```

Comment:
Would it be more appropriate to use a `Set<String>` since the values should be unique?